geoips/compare_outputs.py

```diff
@@ -214,95 +214,76 @@ def images_match(output_product, compare_product, fuzz="5%"):
     bool
         Return True if images match, False if they differ
     """
-    out_diffimg = get_out_diff_fname(compare_product, output_product)
     exact_out_diffimg = get_out_diff_fname(
         compare_product, output_product, flag="exact_"
     )
+    from PIL import Image
+    import numpy as np
+    from pixelmatch.contrib.PIL import pixelmatch
+
+    LOG.info("**Comparing output_product vs. compare product")
+    # Open existing images.
+    out_img = Image.open(output_product)
+    comp_img = Image.open(compare_product)
+    diff_img = Image.new(mode="RGB", size=comp_img.size)
+    # Compute the pixel diff between the two images.
+    if np.array(comp_img).shape == np.array(out_img).shape:
+        diff_arr = np.abs(np.array(comp_img) - np.array(out_img))
+    # If shapes of arrays do not match, pixel diff can not be performed.
+    # Print the names of the two images and associated shapes, and return False.
+    else:
+        LOG.interactive("    ***************************************")
+        LOG.interactive("    *** BAD Images NOT match exactly, different sizes ***")
+        LOG.interactive(
+            "    ***   output_product: %s %s ***",
+            np.array(out_img).shape,
+            output_product,
+        )
+        LOG.interactive(
+            "    ***   compare_product: %s %s ***",
+            np.array(comp_img).shape,
+            compare_product,
+        )
+        LOG.interactive("    ***************************************")
+        return False
```

Comment:
I added a "return False" when the sizes don't match - otherwise was raising an exception and not continuing, we just want the non-zero return.